source/Waves/SpectrumKin.cpp

```diff
@@ -57,25 +57,22 @@ SpectrumKin::getWaveKin(vec3 pos,
 	    amplitudes * (omegas * t - kValues * distances + phases).sin();
 	const Eigen::ArrayX<real> cos_waves =
 	    amplitudes * (omegas * t - kValues * distances + phases).cos();
-	const real surface_height = sin_waves.sum();
 
-	if (pos.z() > surface_height) {
-		if (zeta) {
-			*zeta = surface_height;
-		}
-		if (vel) {
-			*vel = vec3::Zero();
-		}
-		if (acc) {
-			*acc = vec3::Zero();
-		}
-		return;
-	}
+	const real surface_height = sin_waves.sum();
 	const real bottom = actualDepth;
 	const real actual_depth = surface_height - bottom;
 
-	const real stretched_z =
+	real stretched_z =
 	    (-avgDepth * (pos.z() - bottom)) / actual_depth + avgDepth;
+
+	// If the point is above the water surface, return the values at the water
+	// surface. This is important for situations where a point is out of the
```

Comment:
Why is this important (or correct)?

Reply:
You can have a situation like a line floating on the water where the nodes around a segment may be above the water surface (by a small amount), but the segment still has a significant submergence, then surface currents will create drag on the segment in an amount proportional to the submergence and surface water velocity.

If the line nodes are told there is zero water velocity, then there will be no drag forces applied to that segment. 
If the line nodes are told the velocity at the water surface, then the segment will have drag forces calculated in a way that at least approximates reality because lines account for submergence in their drag calculations.